### ISSUE TITLE
simplify dartpad - remove the ability to write and execute html in the snippets

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/dart-pad#820]
We could look at simplifying dartpad (both the codebase and the UI), if we removed support for creating and running html snippets. Writing and running just dart code covers ~90% of the actual uses, and provides the most benefit in terms of teaching and exploring the language.

**Comment by montyr75** [comment]
As the author of a number of web-based Dart tutorials that use DartPad as an easy option to expose people to Dart and its potential, I would be against removing HTML/CSS support. Of course, I do love the idea of adding more options for user interaction, though.

**Comment by bigntallmike** [comment]
I agree, removing the HTML component drastically reduces the functionality of the product.

**Comment by kwalrath** [comment]
I use only the non-HTML/CSS features of DartPad, and as @devoncarew said, that's matches 90% of actual uses. If DartPad supported packages, I might use it for more, but I tend to use a full IDE when I need more. But maybe we're missing important use cases.

@montyr75 and @bigntallmike can you tell us how you use the HTML/CSS features? Links and numbers would be a big help to us.

**Comment by montyr75** [comment]
Personally, I only use DartPad for trying out quick ideas, and those don't usually require HTML/CSS, but since the demise of the old Dart Editor, DartPad has become the only crazy-simple way to get a total beginner exposed to Dart. Many of the web-based tutorials on [Dart Academy](https://dart.academy) direct users to use DartPad to build simple web apps. I even have an interactive tutorial system using an embedded DartPad, and the lessons don't currently make use of HTML/CSS, but they will soon, especially because at the moment there is no other way to facilitate user input in DartPad (no `dart:io`).

- [Dart Academy Boot Camp: Start Programming with Dart](https://dart.academy/dart-academy-boot-camp-start-programming-with-dart/)
- [Web Games with Dart and the HTML5 Canvas: Snake](https://dart.academy/web-games-with-dart-and-the-html5-canvas/)
- [Web Games with Dart: Hangman](https://dart.academy/web-games-with-dart-hangman/)
- [Web Games with Dart: Ultimate Tic-Tac-Toe](https://dart.academy/web-games-with-dart-ultimate-tic-tac-toe/)

These pages get a few hundred views a day, though it's hard to say how many of those readers are actually using DartPad to do the tutorials. Flutter is driving a lot of interest in Dart, though, so the number of daily views is trending up.

Even when I do Dart presentations, I normally walk new users through the process of building simple web apps with DartPad.

In summary, the HTML/CSS support in DartPad is a great tool for quickly exposing new users to the Dart ecosystem and building web apps with Dart, even if people aren't using it for serious work very often.

**Comment by kwalrath** [comment]
> I even have an interactive tutorial system using an embedded DartPad, and the lessons don't currently make use of HTML/CSS, but they will soon, especially because at the moment there is no other way to facilitate user input in DartPad (no `dart:io`).

I believe they plan to add a way to simulate user input: dart-lang/dart-pad#819 is the relevant issue.

**Comment by montyr75** [comment]
Yes, the simple I/O problem is likely being addressed, which is fantastic, but removing HTML/CSS support would severely limit the potential of the more advanced lessons on the interactive tutorial's roadmap. I believe Dart adoption is slowed by how badly its online tools compare to what's available for JavaScript and TypeScript, which both have full-blown IDEs available right in the browser, featuring support for HTML/CSS and even NPM module installation in some cases. It's my view that DartPad's features should be greatly expanded, not cut back, as these kinds of tools are invaluable from the perspective of educating new developers.

**Comment by timsneath** [comment]
I wonder if we need a couple of different 'modes' for DartPad. From a novice Flutter developer's perspective, having HTML and CSS tabs is very confusing. On the other hand, I understand the use case of a web developer using Dart. 

We could theoretically have a flag to disable (or enable) web input, without removing it altogether. For example, https://dartpad.dartlang.org/33706e19df021e52d98c?web=0

**Comment by jcollins-g** [comment]
Related to #860

**Comment by devoncarew** [comment]
closing in favor of #860
